### PR TITLE
Fix small issues around delete/undo

### PR DESCRIPTION
**Pull request by juliushaertl**
- fix: Only query boards not marked for deletion unless we want to undo
- fix: Show proper error page when board was not found
- fix: Properly undo card deletion in the frontend

### CODE REVIEW

File: src/components/board/Board.vue [hunk 1]
```diff
@@ -24,12 +24,19 @@
 	<div class="board-wrapper">
 		<Controls :board="board" />
 
+		{{ loading }}
```
Comment (luka-nextcloud) [inline src/components/board/Board.vue:27]: Is it redundant?

File: lib/Service/CardService.php [hunk 2]
```diff
@@ -308,6 +308,14 @@ public function update($id, $title, $stackId, $type, $owner, $description = '',
 		if ($archived !== null && $card->getArchived() && $archived === true) {
 			throw new StatusException('Operation not allowed. This card is archived.');
 		}
+
+		if ($card->getDeletedAt() !== 0) {
+			if ($deletedAt === null) {
+				// Only allow operations when restoring the card
+				throw new StatusException('Operation not allowed.');
```
Comment (luka-nextcloud) [inline lib/Service/CardService.php:315]: This would be better: `Operation not allowed. This card was deleted.`

**Review by luka-nextcloud — APPROVED** [review]

**Comment by juliushaertl** [comment]
/backport a05d64250fbe602cbc2f1167a6c82381cf25bdb6 to stable26

**Comment by juliushaertl** [comment]
/backport a05d64250fbe602cbc2f1167a6c82381cf25bdb6 to stable25

**Comment by juliushaertl** [comment]
/backport a05d64250fbe602cbc2f1167a6c82381cf25bdb6 to stable24

**Comment by juliushaertl** [comment]
/backport a05d64250fbe602cbc2f1167a6c82381cf25bdb6 to stable23

**Pull request merged**